### ISSUE TITLE
[Logs] Handle top level keys in attributes.

### ISSUE THREAD

**Issue by nityanandagohain** [SigNoz/signoz#2719]
When the otel log contains top level keys in attributes. Ex:- `body` is a top level key but if it is again present in attributes

```
.....
body: " this is a body"
"attributes": {
    "body": " this is another body"
   .........
}

```

As of now we don't have a proper way to handle these cases, now there will be two body attributes in the attributes response api and the query parser will also not work properly for body attribute.

**Comment by srikanthccv** [comment]
@nityanandagohain I forgot our discussion over the huddle. Can you write a clear description here?

**Comment by nityanandagohain** [comment]
@srikanthccv  have update the description

**Comment by srikanthccv** [comment]
What will be the response to the client in this case for keys API? Isn't `IsColumn` and `Type` enough to distinguish whether one is a column and the other is an attribute? I assume they will be shown as `body` and `tag_body` on the frontend?

**Comment by nityanandagohain** [comment]
Yeah, but as of now the the `body` key is also set to tag https://github.com/SigNoz/signoz/blob/1ded475b371f4d0a01874e3c195980f242a6ba4d/pkg/query-service/constants/constants.go#L280

**Comment by nityanandagohain** [comment]
Got, let me add that change.

**Comment by srikanthccv** [comment]
I believe this multiple occurrences of same key is handled in some parts of query builder such as group by so there will be some work needs to be done on the frontend for this.

**Comment by nityanandagohain** [comment]
Yeah, but this can a first level of fix that I can add so that the FE doesn't crash.

**Comment by nityanandagohain** [comment]
Issue 1:- 

When the attributes has a key which is similar to a top level key

```jsx
{"ts": 10233..., "level": "info", "body": "asdasd"}
----------------
.....
body: " this is a body"
"attributes": {
    "body": " this is another body"
   .........
}
```

Issue 2:- When the log has nested keys and we want to parse them

```jsx
{"ts": 10233..., "level": "info", "body": "{\"error\":\"my error\", \"upstream\": {\"error\": \"my error 2\"}}"}
.....
"attributes": {
    "data": {
	"error": "My error 1",
	"upstream": {
	        "error": "my error 2",
	}
   .........
}
```

To solve this we can go with the following approaches

- Add  a prefix to each of these fields in the backed even while creating materialised columns.
    
    The materialized column for attribute body will be `attribute_body`
    
- Add the nested prefix for nested ones
```
attribute_error
attribute_upstream_error
```

**Comment by nityanandagohain** [comment]
* Adding prefix i.e `attribute` has to be done in the `clickhouselogsexporter`.
* to add parent name it has be done in the pipelines , either manual config/ pipelines UI/ JSON UI for facets

**Comment by nityanandagohain** [comment]
Fixed with release v0.27.0
